core/domain/base_model_validators.py

```diff
@@ -84,12 +88,57 @@ def __init__(
             model_class: ClassObject. The external model class.
             model_ids: list(str). The list of external model ids to fetch the
                 external models.
+
+        Raises:
+            Exception. This class was used instead of using the
+                UserSettingsModelFetcherDetails for the UserSettingsModel.
         """
+        if model_class == user_models.UserSettingsModel:
+            raise Exception(
+                'Please use the UserSettingsModelFetcherDetails ' +
```

Comment:
'When fetching instances of UserSettingsModel, please use UserSettingsModelFetcherDetails instead of ExternalModelFetcherDetails.'

Reply:
Done.